File: Factory/FactoryMethod/Java/src/PizzaStore/ChicagoPizzaStore.java
package eu.svetlin.design.patterns.factory.factorymethod.PizzaStore;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.ChicagoStyle.ChicagoStyleCheesePizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.ChicagoStyle.ChicagoStyleClamPizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.ChicagoStyle.ChicagoStylePepperoniPizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.ChicagoStyle.ChicagoStyleVeggiePizza;
import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class ChicagoPizzaStore extends PizzaStore{
    @Override
    Pizza createPizza(String type) {
        if(type.equals("cheese")){
            return new ChicagoStyleCheesePizza();
        }

        if(type.equals("veggie")){
            return new ChicagoStyleVeggiePizza();
        }

        if(type.equals("clam")){
            return new ChicagoStyleClamPizza();
        }

        if(type.equals("pepperoni")){
            return new ChicagoStylePepperoniPizza();
        }

        return null;
    }
}

File: Factory/FactoryMethod/TypeScript/src/PizzaStore/ChicagoPizzaStore.ts
import Pizza from '../Pizza/Pizza';
import PizzaStore from './PizzaStore';
import ChicagoStyleCheesePizza from '../Pizza/ChicagoStyle/ChicagoStyleCheesePizza';
import ChicagoStyleClamPizza from '../Pizza/ChicagoStyle/ChicagoStyleClamPizza';
import ChicagoStylePepperoniPizza from '../Pizza/ChicagoStyle/ChicagoStylePepperoniPizza';
import ChicagoStyleVeggiePizza from '../Pizza/ChicagoStyle/ChicagoStyleVeggiePizza';

class ChicagoPizzaStore extends PizzaStore {
  createPizza(type: string): Pizza {
    if (type === 'cheese') {
      return new ChicagoStyleCheesePizza();
    }

    if (type === 'veggie') {
      return new ChicagoStyleVeggiePizza();
    }

    if (type === 'clam') {
      return new ChicagoStyleClamPizza();
    }

    if (type === 'pepperoni') {
      return new ChicagoStylePepperoniPizza();
    }

    return null;
  }
}

export default ChicagoPizzaStore;
